docs/conf.py

```diff
@@ -91,6 +91,11 @@ def prose_list(items):
 .. |authors| replace:: {author}
 """
 
+# Don't check for type references in docstrings.
+nitpick_ignore_regex = [
+    ('py:.*', '.*'),
+]
+
```

Comment:
This smells stinky to me. It's very broad, and the error messages we're getting (as pasted in the commit message) don't seem to make sense. For example:

```
docstring of augur.align.prepare:1: WARNING: py:class reference target not found: The existing alignment filename
```

Why is the string `The existing alignment filename` being used as a _reference target_ (e.g. a link _destination_)?

Something's fishy here. Is the `sphinx.ext.napoleon` conversion (63297d4ade41bd8592753d58a378663289efd947) doing something bogus? Are a bunch of our docstrings using the wrong syntax for describing parameters?

I think it's worth tracking down what's going on here, as I suspect papering over it now will only cause us more headache in the future.

Reply:
I've replaced 800557793fe9521841b61a4da8d4c624f109c8eb with several fixes: 0c3cb34c20648f7122b749eec399d4d5cf870203...4375322483eb5fa9ec42aa4d7239115a89970a97

This ended up being a doozy, but I learned lots about how these docs are generated!